consensus/config/src/committee.rs

```diff
@@ -162,7 +160,7 @@ impl AuthorityIndex {
 }
 
 impl AuthorityIndex {
-    pub fn new_for_test(index: u32) -> Self {
+    pub fn new(index: u32) -> Self {
```

Comment:
It would be preferable to disallow creating AuthorityIndex outside of Committee, which will prevent accidentally creating invalid AuthorityIndex. It seems we can still call `AuthorityIndex::new_for_test()` in `elect_leader()`, or are there other non-test callsites?